plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/DatafusionEngine.java

```diff
@@ -173,7 +178,7 @@ public Map<String, Object[]> executeQueryPhase(DatafusionContext context) {
 
         try {
             DatafusionSearcher datafusionSearcher = context.getEngineSearcher();
-            long streamPointer = datafusionSearcher.search(context.getDatafusionQuery(), datafusionService.getTokioRuntimePointer());
+            long streamPointer = datafusionSearcher.search(context.getDatafusionQuery(), datafusionService.getTokioRuntimePointer(), datafusionService.getMemoryPoolPtr());
```

Comment:
This will keep extending for every new ptr, lets try using a class of all necessary datafusion ptrs.

Reply:
Updated.